independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -297,7 +297,7 @@ public Optional<ResourceClass> createEndpoints(ClassInfo classInfo, boolean cons
                 }
                 clazz.setPath(sanitizePath(path));
             }
-            if (factoryCreator != null && !classInfo.isInterface()) {
+            if (factoryCreator != null && !classInfo.isInterface() && !classInfo.isAbstract()) {
```

Comment:
This change is specifically there to fix `SubResourceAmbiguousInjectTest`. The `isInterface()` test was added recently to fix the issue and I think the `isAbstract()` makes also sense but happy to be proven wrong.